DataStructures/DynamicStacks/UnitTestDynamicStack.java

```diff
@@ -17,7 +17,10 @@ public static void main(String[] args){
         System.out.println("\n\n\n");
 
 
-        Node nd = new Node().withPayload(5).withNext();
+        Node nd = new Node()
+            .withPayload(5)
+            .withNext(null);
```

Comment:
What are you validating here? I would suggest to validate:
* That `nd` is not null
* That `nd`'s payload is indeed 5
* That `nd`'s next is null

Also add something like this:
```java
Node nd2 = new Node()
    .withPayload(3)
    .withNext(nd);
```
And do the same validations, except checking that `nd2`'s next is equal to `nd`

Reply:
I have made a method for this and will have it in the next commit.